website/project/decorators.py

```diff
@@ -247,6 +247,30 @@ def wrapped(*args, **kwargs):
 must_be_contributor_or_public_but_not_anonymized = _must_be_contributor_factory(include_public=True, include_view_only_anon=False)
 
 
+def must_be_contributor_even_if_public(func):
```

Comment:
Side note: the contributors page gets around this by checking that the user has at least `read` permissions, presumably because the `@must_be_contributor` decorator doesn't actually check that the user must be a contributor. This works because public projects allow non-contribs to view, but they don't explicitly have `read` access, the minimum permission granted to a contributor.

This would also be an acceptable workaround, but if you go this route please add a bug ticket indicating that `_must_be_contributor_factory` is flawed.